numba/__init__.py

```diff
@@ -15,6 +15,8 @@
 # Re-export all type names
 from .types import *
 
+from .smart_array import SmartArray
```

Comment:
According to PEP 8, module names should not have underscores (just `smartarray`, therefore).